src/components/scenegraph/MotionCapture.js

```diff
@@ -0,0 +1,592 @@
+      recordingNames: [],
+      selectedRecordingName: '',  // For replay.
+      uploadedRecordingURL: '',  // For uploading to GitHub Gist.
+      waitForButtonPress: true
+    };
+  }
+
+  componentWillUnmount () {
+    // Remove components.
+    sceneEl.removeAttribute('avatar-recorder');
+    sceneEl.removeAttribute('avatar-replayer');
+  }
+
+  onScriptLoad () {
+    const self = this;
+
+    // Set components.
+    sceneEl.setAttribute('avatar-recorder', {
+      autoPlay: false
+    });
+    sceneEl.setAttribute('avatar-replayer', {
+      autoPlay: false,
+      cameraOverride: '[data-aframe-inspector-original-camera]',
+      loop: JSON.parse(localStorage.getItem(LOCALSTORAGE_LOOP))
+    });
+
+    this.recordingdb = sceneEl.systems.recordingdb;
+
+    // Populate recording names, select from last session if any.
+    this.refreshRecordingNames().then(recordingNames  => {
+      var selectedRecording = localStorage.getItem(LOCALSTORAGE_SELECTED_RECORDING);
+      if (selectedRecording && this.state.recordingNames.indexOf(selectedRecording) !== -1) {
+        this.setState({selectedRecordingName: selectedRecording});
+      }
+
+      // Add sample recording if empty.
+      if (!recordingNames.length) {
+        let xhr = new XMLHttpRequest();
+        xhr.addEventListener('load', function () {
+          self.recordingdb.addRecording('* Sample Recording', JSON.parse(this.responseText));
+          self.refreshRecordingNames().then(() => {
+            self.setState({selectedRecordingName: '* Sample Recording'});
+          });
+        });
+        xhr.open('GET', SAMPLE_RECORDING);
+        xhr.send();
+      }
+    });
+  }
+
+  buttonPressRecording = () => {
+    var self = this;
+
+    const textEntity = document.createElement('a-entity');
+    textEntity.setAttribute('text', {
+      align: 'center',
+      color: 'red',
+      value: 'Press any controller or keyboard button to record'
+    });
+    textEntity.setAttribute('position', '0 0 -1');
+
+    // Resume the scene.
+    sceneEl.play();
+
+    // Set original camera back when recording.
+    const cameraEl = sceneEl.querySelector('[data-aframe-inspector-original-camera]');
+    cameraEl.setAttribute('camera', 'active', true);
+    if (cameraEl.getObject3D('replayermesh')) {
+      cameraEl.getObject3D('replayermesh').visible = false;
+    }
+    cameraEl.appendChild(textEntity);
+
+    // Enter VR.
+    sceneEl.enterVR();
+
+    // Start recording when a button is pressed.
+    sceneEl.addEventListener('buttonup', function buttonStart () {
+      textEntity.parentNode.removeChild(textEntity);
+      self.countdownRecording();
+      sceneEl.removeEventListener('buttonup', buttonStart);
+    });
+    // Start recording when keyboard is pressed.
+    document.addEventListener('keyup', function keyboardStart () {
+      textEntity.parentNode.removeChild(textEntity);
+      self.countdownRecording();
+      sceneEl.removeEventListener('keyup', buttonStart);
```

Comment:
I believe it must be `document` and `keyboardStart`: 
`document.removeEventListener('keyup', keyboardStart);`